### PR TITLE
feat: FDC3 support GENC-224

### PR DESCRIPTION
**Pull request by cistov**
[Review Guidance](https://www.notion.so/genesisglobal/Platform-Code-Review-Process-ace93ba760cc4563b0dfb712e2a88d8a)

📓 &nbsp; **Related JIRA**

https://genesisglobal.atlassian.net/browse/GENC-224

🤔 &nbsp; **What does this PR do?**

- WIP

🚀 &nbsp; **Where should the reviewer start?**

Add file / directory pointers.

📑 &nbsp; **How should this be tested?**

```
TESTAPP=prtestapp && \
rm -rf $TESTAPP && \
npx -y @genesislcap/genx@latest init $TESTAPP -x --fdc3 --ref ac/GENC-224  --apiHost 'wss://public-foundation.genesislab.global/gwf/' --no-npm --no-shell --routes "[{\"name\":\"home\",\"tiles\":[{\"title\":\"Grid Tile\",\"type\":\"grid-pro\",\"config\":{\"resourceName\":\"ALL_POSITIONS\"}},{\"title\":\"Form tile\",\"type\":\"smart-form\",\"config\":{\"resourceName\":\"EVENT_COUNTERPARTY_INSERT\"}}]}]" && \
cd $TESTAPP/client && \
npm run bootstrap && npm run dev
```

✅ &nbsp; **Checklist**

- [ ] I have tested my changes.
- [ ] I have added tests for my changes.
- [ ] I have updated the project documentation to reflect my changes.

### CODE REVIEW

File: .genx/templates/route.hbs [hunk 1]
```diff
@@ -2,13 +2,38 @@ import { customElement, FASTElement } from '@microsoft/fast-element';
...
+
+    this.tile1grid1.addEventListener(GRID_READY_EVENT, () => {
+      this.tile1grid1.gridApi.addEventListener(Events.EVENT_ROW_CLICKED, (e) =>
+        this.fireFDC3GridEvent(e),
+      );
+    });
+  }
+
+  fireFDC3GridEvent(payload) {
+    console.log('fdc3 logic goes here', payload);
```
Comment (patrickoneill-genesis) [inline .genx/templates/route.hbs:35]: do we want to get more prescriptive here?

`
            const channel = await getOrCreateChannel('{{route.fdc3.channel}}');
            const payload = e?.node?.data;

            const m: Context = {
              type: '{{route.fdc3.type}}',
              id: payload,
            };

            await channel.broadcast(m);
`

File: .genx/templates/route.hbs [hunk 2]
```diff
@@ -2,13 +2,38 @@ import { customElement, FASTElement } from '@microsoft/fast-element';
...
   }
+
+  {{#if route.fdc3}}
+  public tile1grid1!: ZeroGridPro;
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.tile1grid1.addEventListener(GRID_READY_EVENT, () => {
+      this.tile1grid1.gridApi.addEventListener(Events.EVENT_ROW_CLICKED, (e) =>
```
Comment (patrickoneill-genesis) [inline .genx/templates/route.hbs:28]: we need to check this code block to check if we are in an fdc3 container

`if (window.fdc3)`

File: .genx/templates/route.hbs (hunk 1 above)
Comment (cistov) [reply .genx/templates/route.hbs:35]: Yeah, that's precisely what I set up this branch - so you could implement something like the above

File: .genx/templates/route.hbs (hunk 2 above)
Comment (cistov) [reply .genx/templates/route.hbs:28]: yeah, I've just set up the foundation for this branch - you'll need to apply the actual changes needed on top

**Comment by cistov** [comment]
This was raised just for visiblity

**Pull request closed without merging**